lib/active_storage/migration.rb

```diff
@@ -14,15 +14,16 @@ def change
 
     create_table :active_storage_attachments do |t|
       t.string  :name
-      t.string  :record_gid
+      t.string  :record_type
+      t.integer :record_id
       t.integer :blob_id
 
       t.datetime :created_at
 
-      t.index :record_gid
       t.index :blob_id
-      t.index [ :record_gid, :name ]
-      t.index [ :record_gid, :blob_id ], unique: true
+      t.index [ :record_type, :record_id ]
+      t.index [ :record_type, :record_id, :name ], name: "index_active_storage_attachments_record_and_name"
+      t.index [ :record_type, :record_id, :blob_id ], name: "index_active_storage_attachments_uniqueness", unique: true
```

Comment:
Looks like these can be collapsed into one `[ :record_type, :record_id, :name, :blob_id ], unique: true`. Preserves record/blob uniqueness and handles both `record.foo_attachments` (record type + id + name) and `blob.record` (record type + id).